lib/ansible/modules/windows/win_shortcut.ps1

```diff
@@ -0,0 +1,95 @@
+#!powershell
+# This file is part of Ansible
+#
+# Ansible is free software: you can redistribute it and/or modify
+# it under the terms of the GNU General Public License as published by
+# the Free Software Foundation, either version 3 of the License, or
+# (at your option) any later version.
+#
+# Ansible is distributed in the hope that it will be useful,
+# but WITHOUT ANY WARRANTY; without even the implied warranty of
+# MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+# GNU General Public License for more details.
+#
+# You should have received a copy of the GNU General Public License
+# along with Ansible.  If not, see <http://www.gnu.org/licenses/>.
+
+# WANT_JSON
+# POWERSHELL_COMMON
+
+$ErrorActionPreference = "Stop"
+$params = Parse-Args $args
+$State = Get-AnsibleParam -obj $params -name "state" -failifempty $true
+$TargetFile = Get-AnsibleParam -obj $params -name "src" -failifempty $true
+$ShortcutFile = Get-AnsibleParam -obj $params -name "dest" -failifempty $true
+$Arguments = Get-AnsibleParam -obj $params -name "arguments" -failempty $true
+$Directory =  Get-AnsibleParam -obj $params -name "directory" -failempty $true
+$hotkey = Get-AnsibleParam -obj $params -name "hotkey" -failifempty $true
+$Description = Get-AnsibleParam -obj $params -name "desc" -failifempty $true
+$IconLocation = Get-AnsibleParam -obj $params -name "icon" -failifempty $true
+$windowstyle = Get-AnsibleParam -obj $params -name "windowstyle" -default $null
+$result = New-Object psobject @{
+    changed = $FALSE
+}
+$src = [System.Environment]::ExpandEnvironmentVariables($TargetFile)
+$dest = [System.Environment]::ExpandEnvironmentVariables($ShortcutFile)
+$arguments = [System.Environment]::ExpandEnvironmentVariables($Arguments)
+$directory = [System.Environment]::ExpandEnvironmentVariables($Directory)
+$icon = [System.Environment]::ExpandEnvironmentVariables($IconLocation)
+$description = [System.Environment]::ExpandEnvironmentVariables($Description)
```

Comment:
I think it is worth putting in the documentation that environment variables in the module parameter values will be expanded.